### PR TITLE
Add aggregators

### PR DESCRIPTION
**Pull request by Pierlou**
Side note: I am wondering whether we want to change the behaviour for malformed queries like in [this test](https://github.com/datagouv/api-tabular/blob/f5afe16a47992956fe86ab01ea31e34fcd076e35/tests/test_query.py#L93). Do we want this kind of query to complete and return stuff, or would we rather have it crash explicitly?
Update: answer "yes"

PS: this PR will be followed by one that introduces auth for requests involving aggregators

### CODE REVIEW

File: api_tabular/utils.py [hunk 1]
```diff
@@ -37,47 +106,90 @@ async def get_app_version() -> str:
...
+            if _filter:
+                sorted = sorted or _sorted
+                sql_query.append(_filter)
+        # aggregators are expected to have the syntax `<column_name>__<operator>`
+        elif len(_split) == 1:
+            column, operator = add_aggregator(_split[0])
+            if column:
+                aggregators[operator].append(column)
+        else:
+            raise ValueError
```
Comment (bolinocroustibat) [inline api_tabular/utils.py:125]: Does a `ValueError` raise a HTTP error 400 with `aiohttp`, or would that raise a HTTP 500?

If that raises a HTTP 500, I think we should replace that with `aiohttp.web.HTTPBadRequest`.

File: api_tabular/utils.py [hunk 2]
```diff
@@ -37,47 +106,90 @@ async def get_app_version() -> str:
...
         sql_query.append(f"limit={page_size}")
     if offset >= 1:
         sql_query.append(f"offset={offset}")
-    if not sorted:
+    if not sorted and not aggregators:
         sql_query.append("order=__id.asc")
     return "&".join(sql_query)
 
 
+def get_column_and_operator(argument):
```
Comment (bolinocroustibat) [inline api_tabular/utils.py:148]: Type hints on this method also would be great :)

File: api_tabular/utils.py [hunk 3]
```diff
@@ -37,47 +106,90 @@ async def get_app_version() -> str:
...
+        elif normalized_comparator == "greater":
+            return f"{column}=gte.{value}", False
+        elif normalized_comparator == "strictly_less":
+            return f"{column}=lt.{value}", False
+        elif normalized_comparator == "strictly_greater":
+            return f"{column}=gt.{value}", False
+    return None, False
+
+
+def add_aggregator(argument):
```
Comment (bolinocroustibat) [inline api_tabular/utils.py:184]: Type hints on this method also would be great :)

File: docker-compose.yml [hunk 4]
```diff
@@ -2,14 +2,15 @@ version: '3'
 
 services:
   postgrest:
-    image: "postgrest/postgrest:v10.1.2"
+    image: "postgrest/postgrest:v12.2.3"
```
Comment (bolinocroustibat) [inline docker-compose.yml:5]: Great! Gotta love recent versions.

Although maybe we should do that in a separate PR in case we have some debug to do?

**Review by bolinocroustibat — APPROVED** [review]
A few remarks!

Otherwise LGTM, great work well done!

File: api_tabular/utils.py (hunk 1 above)
Comment (Pierlou) [reply api_tabular/utils.py:125]: It does raise `400` [here](https://github.com/datagouv/api-tabular/blob/f5afe16a47992956fe86ab01ea31e34fcd076e35/api_tabular/app.py#L99) 💥

File: api_tabular/utils.py (hunk 2 above)
Comment (Pierlou) [reply api_tabular/utils.py:148]: Oops thanks for noticing!

File: api_tabular/utils.py (hunk 1 above)
Comment (bolinocroustibat) [reply api_tabular/utils.py:125]: Ah perfect sorry didn't see

File: docker-compose.yml (hunk 4 above)
Comment (Pierlou) [reply docker-compose.yml:5]: If we want it in a separate PR, it will have to be merged upstream because if I understood it well aggregation funcs were not available before v12

Comment (bolinocroustibat) [reply docker-compose.yml:5]: Understood, then I guess that's ok!

**Comment by bolinocroustibat** [comment]
> Side note: I am wondering whether we want to change the behaviour for malformed queries like in [this test](https://github.com/datagouv/api-tabular/blob/f5afe16a47992956fe86ab01ea31e34fcd076e35/tests/test_query.py#L93). Do we want this kind of query to complete and return stuff, or would we rather have it crash explicitly?

For me, I think it should crash explicitely, returning the error if possible. Syntax should be as deterministic as possible.
